### DISCUSSION TITLE
var d3_1 = require("d3");   ^ Error [ERR_REQUIRE_ESM]: require() of ES Module

### DISCUSSION THREAD

**Discussion by JustinBeck** [antoniolago/react-gauge-component#46] [General]
Hello I'm trying to import React-gauge into my Remix App

```
/Users//Code/playground/remix-simple-indie/node_modules/react-gauge-component/dist/lib/GaugeComponent/index.js:38
var d3_1 = require("d3");
           ^
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users//Code/playground/remix-simple-indie/node_modules/d3/src/index.js from /Users//Code/playground/remix-simple-indie/node_modules/react-gauge-component/dist/lib/GaugeComponent/index.js not supported.
Instead change the require of /Users//Code/playground/remix-simple-indie/node_modules/d3/src/index.js in /Users//Code/playground/remix-simple-indie/node_modules/react-gauge-component/dist/lib/GaugeComponent/index.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/Users//Code/playground/remix-simple-indie/node_modules/react-gauge-component/dist/lib/GaugeComponent/index.js:38:12)
    at Object.<anonymous> (/Users//Code/playground/remix-simple-indie/node_modules/react-gauge-component/dist/lib/index.js:6:40)
    at Object.<anonymous> (/Users//Code/playground/remix-simple-indie/build/index.js:2385:44)
```

I saw the special dynamic import for Next. I tried someother things.. but I'm over my head a bit.
 but I'm using Remix so I didn't know if I could import next/dynamic into it without breaking stuff.
I am going to try to install that in a separate projct to see if I can get around this that. 
Is there any other config or other way to remove this error?

**Comment by mbreitkr** [comment]
I'm running into a similar issue with `require`, albeit while running tests with Vitest. I attempt to use an alias to select the _index.d.ts_ file for import instead but it still just defaults to importing other files from package that use `require`. Is there any update or workaround that will fix this?

**Reply by mbreitkr** [reply]
FYI I found an answer that works [here](https://github.com/vitest-dev/vitest/issues/4007#issuecomment-1691368010) with the test deps optimizer. Not sure of the full ramifications using the config, but my tests are happy!

**Comment by Hleyf** [comment]
Rendering the parent component as a client side fixed the issue for me in NextJs.
